src/dev/github/download_pr_list_cli.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { run, createFlagError, Flags } from '@kbn/dev-utils';
+import fs from 'fs';
+import Path from 'path';
+import { savePrsToCsv } from './search_and_save_pr_list';
+
+function getLabelsPath(flags: Flags) {
+  if (typeof flags.path !== 'string') {
+    throw createFlagError('please provide a single --path flag');
+  }
+
+  if (!fs.existsSync(Path.resolve(flags.path))) {
+    throw createFlagError('please provide an existing json file with --path flag');
+  }
+
+  return Path.resolve(flags.path);
+}
+
+export async function downloadPullRequests() {
+  run(
+    async ({ log, flags }) => {
+      const githubToken = process.env.GITHUB_TOKEN;
+
+      if (!githubToken) {
+        throw new Error('GITHUB_TOKEN was not provided.');
+      }
+
+      const labelsPath = getLabelsPath(flags);
+
+      if (typeof flags.dest !== 'string') {
+        throw createFlagError('please provide csv path in --dest flag');
+      }
+
+      const query = flags.query;
+      if (query && typeof query !== 'string') {
+        throw createFlagError('please provide valid string in --query flag');
+      }
+
+      const mergedSince = flags['merged-since'];
+      if (
+        mergedSince &&
+        (typeof mergedSince !== 'string' || !/\d{4}-\d{2}-\d{2}/.test(mergedSince))
+      ) {
+        throw createFlagError(
+          `please provide a past date in 'yyyy-mm-dd' format in --merged-since flag`
+        );
+      }
```

Comment:
@spalger I merged your commit and realised both `query` and `mergedSince` are optional.

I added a check for undefined/null but it still won't eliminate `number` as possible type, so I changed related arguments type to `any` in `savePrsToCsv`. Not sure it is the best solution, but that function has internal check for undefined and it can't be number anyway.

Reply:
When they're optional I use:

```js
const variable = flags.name || undefined;
```

With this an empty string value will be converted to `undefined` which is how you want your "optional" variables to be passed anyway, and you can then just add `if (variable !== undefined && typeof variable !== 'string)`

Reply:
Thank you for the explanation, I pushed it in 0e5436a8b8c5061ce448720dafaea2ccf84defa9